### PR TITLE
 Blobbernauts now die slowly when not near the blob

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
tweak: Blobbernauts now die slowly when not near a blob.
rscadd: Blobbernauts can now see their health and the core health of the overmind that created them.
/:cl:

Blobbernauts are now slightly resistant to spacewind.
Xenobio can no longer get blobbernauts from gold slime cores.
Fixes a bug where the overmind core health display would disappear in 'reduced' HUD mode.
HUD!

### CODE REVIEW

**Comment by Buggy123** [comment]
Seems reasonable but I don't think blob particularly need a nerf right now.

**Comment by newfren** [comment]
blobbernauts definitely do

:+1:

**Comment by AdamElTablawy** [comment]
blob? blob's balanced, atm, if a bit iffy with some of the new chems.

Blobbernauts, however, need a definite nerf.

**Comment by GunHog** [comment]
I hope this will be the last of blobbernaut nerfs. I am saying this even as I lost to them twice in recent days.

**Comment by AdamElTablawy** [comment]
Nah they still need more nerfs.

Player control is already insanely strong. This isn't even that strong.

**Comment by Robustin** [comment]
Cutting max HP makes more sense.

Good blobbernauts rarely go far from the blob anyway.

**Comment by KazeEspada** [comment]
:+1:  Keep those meat shields near the blob.

**Comment by Bawhoppen** [comment]
:-1: stealth xenobio nerf

**Comment by GunHog** [comment]
Why do nauts not have basic wall smash?

**Comment by RemieRichards** [comment]
@GunHog because they were given human consciousness

**Comment by GunHog** [comment]
Were they smashing walls to space or something? I am guessing it would be due to them leaving the blob and smashing into departments to kill scientists, doctors and the like?

**Comment by RemieRichards** [comment]
@GunHog when I initially made blobbernauts they had the highest wallsmash possible (because they were dumb AI) iirc they kept this during the first phase of playernauts.

players can do a lot with smashing walls.

**Comment by ChangelingRain** [comment]
They basically smashed every wall they could see, yeah.
I might give them basic wallsmash back if I decide to increase the off-blob damage, so they can't circle the station, smashing inwards.

**Comment by Robustin** [comment]
Again reducing max HP solves all the issues.

Blobbernauts who stray from the blob don't face a big risk because they can tank so many shots and just run back. The HP decay effectively reduces the max HP, but only if you spend a significant amount of time off blob tiles.

Of course the catch is that renegade nauts who would run off were never the issue, the standard "best practices" blubbernaut would involve camping on the blob and a ridiculous barrier to anyone trying to weld the blob (this alone is a winning tactic) and occasionally pushing out to hit a group while they're busy recharging/healing/whatever.

This won't fix anything about blubbernauts and it adds unintended casualties (xenobio nauts).

Cut max HP by 40-60 and see where they stand. Right now camping nauts are the biggest issue, 30 power with virtually no drawback (I only spawn them from "safe" factories anyway) and if they are on the front lines they can:

1) Shield a vital node with their body until the blob reinforces/regens

2) Essentially DOUBLE-QUADRUPLE the blob's offensive power (blob has to spend 5 tc to do an attack that nauts do for free, nauts pay for themselves after just a few strikes)

3) The traditional front-line of blob fighting was welders. It took a good amount of time for rechargers and emitters to get set up and even then welders still beat lasers for raw damage/second efficiency. Blubbernauts shit allllll over this standard blob fighting tactic. While welding used to be about smart positioning and intelligent attack routes, its now almost suicidal when a naut is up. Before even when the blob would manage to strike you, it couldn't rapidly finish you off or chase you down. Now the blubbernaut can beat a crit to death in seconds (added bonus: free zombies) before someone can drag them out and whereas someone could take a couple blob hits before and still run away, now anything less than 100% move speed will have the naut run you down and kill you. 

All that for 30 goddamn power, insignificant for a blob after a couple minutes. 

Cut max HP or double the cost so that it's the luxury end-game crew stomper that actually exhibits balance. Plus commando nauts are actually hilarious and entertaining, killing them is just bad, bad bad. I love the idea of a naut sneaking around the station and busting into escape and beating up the nerds working RND.

**Comment by KorPhaeron** [comment]
This kills some freedom while not really doing much to nerf them yeah.

At least remove them from xenobio if you are set on doing this.

**Comment by ChangelingRain** [comment]
I have some more directed nerfs in mind for nauts, but I'd rather do them in another PR after some live testing of this.

**Comment by Robustin** [comment]
Well then focus on those instead of busting out the most gameplay-limiting one first.

**Comment by ChangelingRain** [comment]
I'd "focus" on those nerfs, except they're literally two lines to slow nauts down and halve how fast they attack, and honestly I could roll them into this PR except I'd prefer not to.

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 1]
```diff
@@ -200,6 +200,24 @@
...
 
+/mob/living/simple_animal/hostile/blob/blobbernaut/Life()
+	if(..())
+		if(!(locate(/obj/effect/blob) in range(2, src)))
+			adjustHealth(maxHealth*0.025) //take 2.5% maxhealth as damage when not near the blob
+			var/list/viewing = list()
+			for(var/mob/M in viewers(src))
+				if(M.client)
+					viewing += M.client
+			flick_overlay(image('icons/mob/blob.dmi', src, "nautdamage", MOB_LAYER+0.1), viewing, 8)
```
Comment (AnturK) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: This probably could use a message to the blobbernaut why it's dying but i guess they will learn.

File: code/game/gamemodes/blob/overmind.dm [hunk 2]
```diff
@@ -61,6 +61,9 @@
 /mob/camera/blob/update_health_hud()
 	if(blob_core)
 		hud_used.healths.maptext = "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='#e36600'>[round(blob_core.health)]</font></div>"
+		for(var/mob/living/simple_animal/hostile/blob/blobbernaut/B in blob_mobs)
+			if(B.hud_used && B.hud_used.blobpwrdisplay)
+				B.hud_used.blobpwrdisplay.maptext = "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='#82ed00'>[round(blob_core.health)]</font></div>"
```
Comment (phil235) [inline code/game/gamemodes/blob/overmind.dm]: why are we updating the blobberanut's hud when we update the blob's hud?

Comment (AnturK) [inline code/game/gamemodes/blob/overmind.dm]: I assume to save on backref on the blobbernaut. Probably could be moved to blobbernaut hud.

File: code/game/gamemodes/blob/blobs/blob_mobs.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: Well, there's this, a changelog, the HUD will show health loss, the spawn message says something about dying when off the blob, I think they'll get it.

File: code/game/gamemodes/blob/overmind.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/game/gamemodes/blob/overmind.dm]: I didn't see another good way to update the naut's hud whenever the core health changed(without odd delays) so this seemed like a decent solution.

Comment (phil235) [inline code/game/gamemodes/blob/overmind.dm]: Oh I had misread the line. But using the blobpwrdisplay var to display the blob core's HEALTH is pretty confusing...

Comment (ChangelingRain) [inline code/game/gamemodes/blob/overmind.dm]: yeah it's technically cheating but I didn't want to add another var to the base hud

**Pull request merged**